core/drivers/pm/sam/at91_pm.c

```diff
@@ -311,13 +325,27 @@ TEE_Result atmel_pm_suspend(uintptr_t entry, struct sm_nsec_ctx *nsec)
 
 	sm_save_unbanked_regs(&nsec->ub_regs);
 
+	/*
+	 * In order to run code for low-power out of SRAM without abort,
+	 * configure regions with write permission with not forced to
+	 * XN (Execute-never) attribute.
+	 */
+	if (read_id_pfr1() & IDPFR1_VIRT_MASK) {
+		sctlr = read_sctlr();
+		write_sctlr(sctlr & ~SCTLR_WXN);
+	}
+
 	if (soc_pm.mode == AT91_PM_BACKUP) {
 		res = at91_enter_backup();
 	} else {
 		at91_suspend_sram_fn(&soc_pm);
 		res = TEE_SUCCESS;
 	}
 
+	/* Restore the XN attribute */
+	if (read_id_pfr1() & IDPFR1_VIRT_MASK)
```

Comment:
How about checking if `SCTLR_WXN` is set in the `sctlr` variable and only write it then?
I believe the TLB should be invalidated if `SCTLR_WXN` is changed in `SCTLR` here too.